src/rapids_dependency_file_generator/rapids_dependency_file_generator.py

```diff
@@ -127,8 +131,38 @@ def make_dependency_file(
                 "dependencies": dependencies,
             }
         )
-    if file_type == str(OutputTypes.REQUIREMENTS):
+    elif file_type == str(OutputTypes.REQUIREMENTS):
         file_contents += "\n".join(dependencies) + "\n"
+    elif file_type == str(OutputTypes.PYPROJECT):
+        # This file type needs to be modified in place instead of built from scratch.
+        with open(os.path.join(output_dir, name)) as f:
+            file_contents = tomlkit.load(f)
+
+        toml_deps = tomlkit.array()
+        for dep in dependencies:
+            toml_deps.add_line(dep)
+        toml_deps.add_line(indent="")
+        toml_deps.comment(
+            f"This list was generated by `{cli_name}`. To make changes, edit "
+            f"{relative_path_to_config_file} and run `{cli_name}`."
+        )
+
+        # Recursively descend into subtables like "[x.y.z]", creating tables as needed.
```

Comment:
While valid TOML, are nested tables like this permitted per the pyproject.toml schema?

Reply:
Yes, certain specific ones are (e.g. project.optional-dependencies). Which ones we allow can be controlled at the validation stage as we're discussing in the other thread.

Reply:
Right, I hadn't realized that this is already two nestings because of the top-level "project".